src/test/java/com/iexec/resultproxy/jwt/JwtServiceTests.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.resultproxy.jwt;
+
+import com.iexec.common.security.SignedChallenge;
+import com.iexec.common.utils.FileHelper;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.SignatureException;
+import io.jsonwebtoken.UnsupportedJwtException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.web3j.crypto.Credentials;
+import org.web3j.crypto.ECKeyPair;
+import org.web3j.crypto.Keys;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.SecureRandom;
+import java.util.Date;
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.iexec.resultproxy.jwt.JwtService.KEY_SIZE;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+class JwtServiceTests {
+
+    @TempDir
+    private static Path tmpDir;
+    private JwtConfig jwtConfig;
+    @Mock
+    private JwtRepository jwtRepository;
+    private JwtService jwtService;
+    private String walletAddress;
+    private static final byte[] badJwtKey = SecureRandom.getSeed(KEY_SIZE);
+
+    private String getWalletAddress() {
+        try {
+            ECKeyPair ecKeyPair = Keys.createEcKeyPair();
+            return Credentials.create(ecKeyPair).getAddress();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @BeforeEach
+    void init() throws IOException {
+        jwtConfig = new JwtConfig(String.join(File.separator, tmpDir.toString(), ".key"));
+        MockitoAnnotations.openMocks(this);
+        walletAddress = getWalletAddress();
+        jwtService = new JwtService(jwtConfig, jwtRepository);
+    }
+
+    @Test
+    void shouldNotCreateServiceWhenCorruptKey() throws IOException {
+        Path keyFilePath = Path.of(jwtConfig.getKeyPath());
+        Files.deleteIfExists(keyFilePath);
+        FileHelper.createFileWithContent(jwtConfig.getKeyPath(), UUID.randomUUID().toString());
+        assertThrows(IllegalArgumentException.class,
+                () -> new JwtService(jwtConfig, jwtRepository));
+        Files.deleteIfExists(keyFilePath);
+    }
```

Comment:
Could you please add a short comment to explain why this key is corrupted? Something like: "The key should be Base64 whereas we use a UUID there."

Reply:
Renamed the test method in 3cf9b30